test/OpenFeature.Contrib.Providers.Statsig.Test/EvaluationContextExtensionsTests.cs

```
@@ -0,0 +1,112 @@
+                .Set(EvaluationContextExtensions.CONTEXT_COUNTRY, country)
+                .Set(EvaluationContextExtensions.CONTEXT_EMAIL, email)
+                .Set(EvaluationContextExtensions.CONTEXT_IP, ipAddress)
+                .Set(EvaluationContextExtensions.CONTEXT_LOCALE, locale)
+                .Set(EvaluationContextExtensions.CONTEXT_USER_AGENT, userAgent).Build();
+
+            // Act
+            var statsigUser = evaluationContext.AsStatsigUser();
+
+            // Assert
+            Assert.NotNull(statsigUser);
+            Assert.Equal(statsigUser.AppVersion, appVersion);
+            Assert.Equal(statsigUser.Country, country);
+            Assert.Equal(statsigUser.Email, email);
+            Assert.Equal(statsigUser.IPAddress, ipAddress);
+            Assert.Equal(statsigUser.Locale, locale);
+            Assert.Equal(statsigUser.UserAgent, userAgent);
+        }
+
+        [Theory]
+        [AutoData]
+        public void AsStatsigUser_ShouldMapPrivateData(string key, string value)
+        {
+            var privateProperties = new Dictionary<string, Value>() { { key, new Value(value) } };
+
+            // Arrange
+            var evaluationContext = EvaluationContext.Builder().Set(EvaluationContextExtensions.CONTEXT_PRIVATE_ATTRIBUTES, new Structure(privateProperties)).Build();
+
+            // Act
+            var statsigUser = evaluationContext.AsStatsigUser();
+
+            // Assert
+            Assert.NotNull(statsigUser);
+            Assert.True(statsigUser.PrivateAttributes.TryGetValue(key, out var mappedValue));
+            Assert.Equal(value, (mappedValue as Value)?.AsString);
+        }
+
+        [Fact]
+        public void AsStatsigUser_ShouldHandleNullEvaluationContext()
+        {
+            // Arrange
+            EvaluationContext evaluationContext = null;
+
+            // Act
+            var statsigUser = evaluationContext.AsStatsigUser();
+
+            // Assert
+            Assert.Null(statsigUser);
```

Curious as to why this is desirable. If the eval context is null, why isn't it more helpful to throw rather than to return a null value for `statsigUser`?

---

The Statsig API handles the null user edge case (by returning false for everything) and I haven't thought of any reason as to diverge from this. But I certainly can throw an exception instead if you think that is preferable.